pylutron_caseta/smartbridge.py

```diff
@@ -989,6 +1023,70 @@ def _process_occupancy_group(self, occgroup):
             name=f"{self.areas[occgroup_area_id]['name']} Occupancy",
         )
 
+    async def _load_ra3_occupancy_groups(self):
+        """Load the devices from the bridge and filter for occupancy sensors."""
+        _LOG.debug("Finding occupancy sensors from bridge")
+        occdevice_json = await self._request(
+            "ReadRequest", "/device?where=IsThisDevice:false"
+        )
+        if occdevice_json.Body is None:
+            return
+
+        occdevices = occdevice_json.Body.get("Devices", {})
+        for occdevice in occdevices:
+            if occdevice["DeviceType"] in RA3_OCCUPANCY_SENSOR_DEVICE_TYPES:
+                self._process_ra3_occupancy_group(occdevice)
+
+    def _process_ra3_occupancy_group(self, occdevice):
+        """Process ra3 occupancy group."""
+        occdevice_id = id_from_href(occdevice["href"])
+        occsensor_ids = []
+        associated_area = occdevice["AssociatedArea"]
+        occgroup_area_id = id_from_href(associated_area["href"])
+
+        if occgroup_area_id not in self.areas:
+            _LOG.error(
+                "Unknown parent area for occupancy group %s: %s",
+                occdevice_id,
+                occgroup_area_id,
+            )
+            return
+
+        if occgroup_area_id in self.occupancy_groups:
+            _LOG.debug(
+                "Occupancy area %s already registered, append sensor %s",
+                occgroup_area_id,
+                occdevice_id,
+            )
+            self.occupancy_groups[occgroup_area_id]["sensors"].append(occdevice_id)
+            return
+
+        occsensor_ids.append(occdevice_id)
+
+        self.occupancy_groups.setdefault(
```

Comment:
This function seems like maybe it was partially refactored.

At the top of the function, a variable `occsensor_ids` is initialized to `[]`. This variable is only read once and only after it is updated to `[occdevice_id]`. It could be replaced with just `[occdevice_id]`.

This code here uses `self.occupancy_groups.setdefault(occgroup_area_id, …)`, but the code just before it does an early return if `occgroup_area_id in self.occupancy_groups`. So this code could be simplified to `self.occupancy_groups[occgroup_area_id] = { occupancy_group_id: occgroup_area_id, status: OCCUPANCY_GROUP_UNKNOWN, sensors: occsensor_ids, name: f"{self.areas[occgroup_area_id]['name']} Occupancy" }`.

Alternatively, the return value of `self.occupancy_groups.setdefault(occgroup_area_id, { …, sensors: [], })` could be stored and then `whatever.sensors.append(occ_device_id)` would add the sensor to the list, but then the logging would be different.

Reply:
Yes, I agree.  I've removed the unused variable occsensor_ids, and taken your advice with returning the value of self.occupancy_groups.setdefault(..) and appending the sensor.  The early check and return has been removed.